newIDE/app/src/CompactPropertiesEditor/index.js

```diff
@@ -0,0 +1,791 @@
+            //   )
+            // }
+          />
+        );
+      } else if (field.valueType === 'color') {
+        return 'TODO';
+        // const { setValue } = field;
+        // return (
+        //   <Column key={field.name} expand noMargin>
+        //     <ColorField
+        //       id={field.name}
+        //       floatingLabelText={getFieldLabel({ instances, field })}
+        //       helperMarkdownText={getFieldDescription(field)}
+        //       disableAlpha
+        //       fullWidth
+        //       color={getFieldValue({ instances, field })}
+        //       onChange={color => {
+        //         const rgbString =
+        //           color.length === 0 ? '' : rgbOrHexToRGBString(color);
+        //         instances.forEach(i => setValue(i, rgbString));
+        //         _onInstancesModified(instances);
+        //       }}
+        //     />
+        //   </Column>
+        // );
+      } else if (field.valueType === 'enumIcon') {
+        const value = getFieldValue({ instances, field });
+        return (
+          <IconButton
+            key={field.name}
+            id={field.name}
+            size="small"
+            tooltip={getFieldLabel({ instances, field })}
+            selected={field.isHighlighted(value)}
+            onClick={event => {
+              instances.forEach(i => field.setValue(i, !value));
+              _onInstancesModified(instances);
+            }}
+          >
+            {field.renderIcon(value)}
+          </IconButton>
+        );
+      } else if (field.valueType === 'textarea') {
+        return 'TODO';
+        // const { setValue } = field;
+        // return (
+        //   <SemiControlledTextField
+        //     key={field.name}
+        //     id={field.name}
+        //     onChange={text => {
+        //       instances.forEach(i => setValue(i, text || ''));
+        //       _onInstancesModified(instances);
+        //     }}
+        //     value={getFieldValue({ instances, field })}
+        //     floatingLabelText={getFieldLabel({ instances, field })}
+        //     floatingLabelFixed
+        //     helperMarkdownText={getFieldDescription(field)}
+        //     multiline
+        //     style={styles.field}
+        //   />
+        // );
+      } else if (field.valueType === 'resource') {
+        return 'TODO';
+      } else {
+        const {
+          // TODO: Still support onEditButtonClick & onEditButtonBuildMenuTemplate ?
+          // onEditButtonBuildMenuTemplate,
+          // onEditButtonClick,
+          setValue,
+          onClickEndAdornment,
+        } = field;
+        return (
+          <CompactSemiControlledTextField
+            value={getFieldValue({
+              instances,
+              field,
+              defaultValue: '(Multiple values)',
+            })}
+            id={field.name}
+            // floatingLabelText={getFieldLabel({ instances, field })}
+            // helperMarkdownText={getFieldDescription(field)}
```

Comment:
if those 2 fields are not part of the new compact text field, is it useful to keep those comments? (same question for all below)